### PR TITLE
Replace FileDownloader with OkDownload seamlessly

### PR DESCRIPTION
**Pull request by rantianhua**

### CODE REVIEW

File: okdownload-filedownloader/src/main/java/com/liulishuo/filedownloader/BaseDownloadTask.java [hunk 1]
```diff
@@ -225,6 +225,7 @@
      * to the global queue to make this task is ready to be assembled by the queue which makes up of
      * the same listener task, just invoke {@link InQueueTask#enqueue()}.
      */
+    @Deprecated
```
Comment (Jacksgong) [inline okdownload-filedownloader/src/main/java/com/liulishuo/filedownloader/BaseDownloadTask.java]: `deprecated`后面给出建议的替代方法吧。

File: okdownload-filedownloader/src/main/java/com/liulishuo/filedownloader/DownloadTaskAdapter.java [hunk 2]
```diff
@@ -525,8 +525,7 @@ public void startTaskByRescue() {
 
     @Override
     public Object getPauseLock() {
```
Comment (Jacksgong) [inline okdownload-filedownloader/src/main/java/com/liulishuo/filedownloader/DownloadTaskAdapter.java]: 给下`Nullable`的`annotation`。

File: okdownload-filedownloader/src/main/java/com/liulishuo/filedownloader/DownloadTaskAdapter.java [hunk 3]
```diff
@@ -438,8 +446,7 @@ public boolean isSyncCallback() {
 
     @Override
     public boolean isLargeFile() {
-        // TODO: 2018/7/25
-        return false;
+        return listener != null && listener instanceof FileDownloadLargeFileListener;
```
Comment (Jacksgong) [inline okdownload-filedownloader/src/main/java/com/liulishuo/filedownloader/DownloadTaskAdapter.java]: 不需要`listener != null`，`null instanceof XXX`肯定为`false`

File: okdownload-filedownloader/src/main/java/com/liulishuo/filedownloader/DownloadTaskAdapter.java [hunk 4]
```diff
@@ -128,6 +128,9 @@ public BaseDownloadTask setTag(Object tag) {
 
     @Override
     public BaseDownloadTask setTag(int key, Object tag) {
+        if (key == TAG_KEY) {
```
Comment (Jacksgong) [inline okdownload-filedownloader/src/main/java/com/liulishuo/filedownloader/DownloadTaskAdapter.java]: `TAG_KEY`并没有表达出该参数的作用，需要调整下。

File: okdownload-filedownloader/src/main/java/com/liulishuo/filedownloader/BaseDownloadTask.java (hunk 1 above)
Comment (rantianhua) [reply okdownload-filedownloader/src/main/java/com/liulishuo/filedownloader/BaseDownloadTask.java]: 在注释里面有的。

**Comment by rantianhua** [comment]
@std829 Yes, you need to include it separately in the gradle.
```
com.liulishuo.okdownload:filedownloader:1.0.4
```

**Comment by std829** [comment]
@rantianhua thanks I'll try that! I think at the time I tried that but it wasn't recognising version 1.0.4 for filedownloader

**Pull request merged**